### ISSUE TITLE
Unhandled exception in script

### ISSUE THREAD

**Issue by Monroe07** [DA-Osborne/Spool-Maker#6]
Not working with CURA v5, works on CURA v4.12.1
OS tested, Windows 11.
![Screenshot 2022-05-22 132513](https://user-images.githubusercontent.com/20032550/169716487-cb927265-7482-4b72-8581-1d0e62283502.png)

**Comment by DA-Osborne** [comment]
Hmm, Cura 5 must store its version information in the registry differently to 4. I'll take a look. I haven't tried Cura 5 yet.

**Comment by Monroe07** [comment]
I got it to work with Cura 5, but only with Cura 4.12.1 installed in parallel. Oddly enough, I was able to verify it was loading from Cura 5 by adding a material in Cura 5 and it showed up in SpoolMaker. I'll uninstall Cura 4 and report back if it still works. I'm running a clean install of Windows 11, installed yesterday if it helps troubleshoot.

**Comment by Monroe07** [comment]
With Cura 4 removed, its back to the initial unhandled exception

**Comment by rachael7** [comment]
Given that it is a registry reading issue, could it possibly be related to Windows 11?

**Comment by Monroe07** [comment]
I can test that on another computer that is windows 10 and has never had Cura on it before, but given that SpoolMaker (All 3 release versions) works on windows 11 with Cura 4.12.1 without issue, I'd tend to disagree that it could be an OS issue.

**Comment by Monroe07** [comment]
I have now tested with Cura 5 only (No other versions of Cura has ever been installed on the second PC) on Windows 10, I get the same error, that a file (registry entry) cannot be found. It is identical to the message pictured above.

**Comment by DA-Osborne** [comment]
Ok, so Ultimaker 5.0 does not seem to store any version information/install locations in that registry key anymore. Will see if I can find where it is now storing it (if at all). The reason it may have picked up materials from 4 is because the registry key pertains to the materials stored in Programs Files, not AppData.
![image](https://user-images.githubusercontent.com/43732435/169949978-636f2ea1-7d77-4424-99af-56fa07d400ec.png)
It instead only picks up the latest version in there, v4. If you uninstall everything but v5, there is nothing left and you get the error, as there are no registry keys to load.

**Comment by DA-Osborne** [comment]
Okay this key might be the best option: Computer\HKEY_CURRENT_USER\Software\Microsoft\Windows\CurrentVersion\App Paths\Ultimaker-Cura.exe

**Comment by DA-Osborne** [comment]
Ok I believe I fixed the issue by checking for the v5 key first, then the v4 keys if that is not found. Before I close, can someone check it works for them? (v1.1.3 release)

**Comment by Monroe07** [comment]
Works for me! Thank you!

**Comment by rachael7** [comment]
I just tested the compiled executable on a Windows 10 system with only Cura 4.13.1 installed and I received the error, "Failed to determine Ultimaker Cura install location". I tried the uncompiled Python, with the same result. Looking inside CuraMaterial.py and comparing it to the previous version, I believe there is an oversight on line 72, which reads:

`            curaSysDir = QueryValueEx(curaSysDir, '')[0]`

I believe that line should read:

`            curaSysDir = QueryValueEx(curaSysDirKey, '')[0]`

Making that change and running the uncompiled version again, it seems to work.  Thanks for keeping this wonderful utility up to date!
